arelle/plugin/validate/HMRC/ValidateHmrc.py

```diff
@@ -331,13 +347,54 @@ def _lang(self) -> HmrcLang:
                         return HmrcLang.WELSH
         return HmrcLang.ENGLISH
 
+    def _verifyAuditFacts(self, prefix: str) -> None:
+        """
+        Logs an error when an audited report does not facts tagged with the concepts of "DateAuditorsReport",  "OpinionAuditorsOnEntity"
+        as well as either "NameIndividualAuditor" or the combination of "NameSeniorStatutoryAuditor" and "NameEntityAuditors".
+        :param prefix:
+        :return:
+        """
+        missingConcepts = []
+        dateFacts = self._getFacts(CONCEPT_DATE_AUDITOR_REPORT)
+        if not any(self._checkValidFact(x) for x in dateFacts):
```

Comment:
The `_checkValidFact` helper is a nice addition. I'd even go a step further and add `_checkValidFacts` as will, which retrieves the facts and performs the `any(...)` logic, since it seems to be repeated several times.